tests/roots.py

```diff
@@ -0,0 +1,46 @@
+def root_bisec(f,a=0,b=1,error=10**-7):
+    '''Root of a function using the bisection method
+    
+    The aruments expected are:
+    f: a function with the format double f(double x)
+    a,b: bounds with a separated root.
+    error: precision for the root determination default value 10^-7
+    '''
+    ext=f(a)*f(b)
+    if ext>0:
+        return nul
+    elif ext==0:
+        for t in [a,b]:
+            if f(t)==0:
+                return t
+    else:
+        while abs(a-b)>error:
+            x=(a+b)/2
+            ext=f(a)*f(x)
+            if ext==0:
+                return x
+            elif ext>0:
+                a=x
+            else:
+                b=x
+        return x
+    
+def root_newton(f,df,a,error=10**-7):
+    '''Root of a function using the Newton-Raphson method
+    
+    The aruments expected are:
+    f: a function with the format double f(double x)
+    df: derivative of the function with the same format
+    a: initial approximation of the root.
+    error: precision for the root determination default value 10^-7
+    '''
+    ext=f(a)
+    lx=a
+    if ext==0:
+        return a
+    cond=True
+    while cond:
+        x=lx-f(lx)/df(lx)
+        cond=abs(x-lx)>error
+        lx=x
+    return x
```

Comment:
These tests do not test the fact that the mass matrix works... ;-)

Can you remove these, and add a test that checks if the mass matrix is doing what we expect? Keep in mind that doing

~~~
v*M.dot(v)
~~~
is equivalent to integrating `v**2`. Test that this is true, with a vector of ones, for example. You should get one.